### PR TITLE
[REVIEW] Fix `.str.rsplit`, `.str.split`, `.str.find`, `.str.rfind`, `.str.index`, `.str.rindex` and enable related tests

### PR DESCRIPTION
**Pull request by galipremsagar**
Made fixes as part of enabling disabled tests.

Enabling tests because of the following issues are fixed:
#4380

### CODE REVIEW

File: python/cudf/cudf/tests/test_string.py [hunk 1]
```diff
@@ -1591,7 +1587,6 @@ def test_string_starts_ends(data, pat):
     [
         # TODO, PREM: Uncomment after this issue is fixed
         # '',
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_string.py]: Is this case still broken?

Comment (kkraus14) [inline python/cudf/cudf/tests/test_string.py]: Is this case still broken?

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_string.py]: Yes, I'm trying to see if I can fix the last few of the leftover TODO's by handling in python/cython layer. If not will have to get them resolved in libcudf layer.

File: python/cudf/cudf/core/column/string.py [hunk 2]
```diff
@@ -879,10 +879,11 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
         from cudf._libxx.scalar import Scalar
 
         result_table = cpp_split(self._column, Scalar(pat, "str"), n)
-
         if len(result_table._data) == 1:
             if result_table._data[0].null_count == len(self._parent):
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: `self._parent` can be `None`:
```suggestion
            if result_table._data[0].null_count == len(self._column):
```

File: python/cudf/cudf/core/column/string.py [hunk 3]
```diff
@@ -879,10 +879,11 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
...
                 result_table = []
+            elif self._parent.null_count == len(self._parent):
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py]: `self._parent` can be `None`:
```suggestion
            elif self._column.null_count == len(self._column):
```

**Comment by galipremsagar** [comment]
The recent changes are a cap-gap for #4500

**Pull request merged**